web-app/HeimScripts/heatmap/summary.R

```diff
@@ -151,12 +257,17 @@ extract_measurements <- function(datasets)
 produce_summary_stats <- function(measurement_tables, phase)
 {
   # construct data.frame to store the results from the summary statistics in
-  result_table <- as.data.frame(matrix(NA, length(measurement_tables),12, 
+  result_table <- as.data.frame(matrix(NA, length(measurement_tables),13, 
                                        dimnames = list(names(measurement_tables), 
-                                                       c("variableLabel","node","subset","totalNumberOfValuesIncludingMissing", "numberOfMissingValues", "min","max","mean", "standardDeviation", "q1","median","q3"))))
+                                                       c("variableLabel","node","subset","totalNumberOfValuesIncludingMissing", "numberOfMissingValues", "numberOfSamples","min","max","mean", "standardDeviation", "q1","median","q3"))))
+  
   # add information about node and subset identifiers
-  result_table$subset <- gsub(".*_","", rownames(result_table))
-  result_table$node <- gsub("_.*","", rownames(result_table))
+  result_table$subset <- gsub(".*_","", rownames(result_table)) #take everything after _
```

Comment:
There can only be one "_" in current implementation, with the current agreements about the format of 'loaded_variables' and 'preprocessed'. The identifier that is used here for gsub() either has format n<node_id>_s<subset_id> in case of 'fetch data' (which is the same as the labels of the data.frames in 'loaded_variables'), or for 'preprocess' it's preprocessed_s<subset_id> (which is determined in get_input_data + split_on_subsets). Should I make it more specific so that it will work for cases where there are multiple underscores?